Firmware/MotorControl/encoder.cpp

```diff
@@ -32,31 +32,46 @@ bool Encoder::apply_config(ODriveIntf::MotorIntf::MotorType motor_type) {
 }
 
 void Encoder::setup() {
-    HAL_TIM_Encoder_Start(timer_, TIM_CHANNEL_ALL);
-    set_idx_subscribe();
-
     mode_ = config_.mode;
 
-    spi_task_.config = {
-        .Mode = SPI_MODE_MASTER,
-        .Direction = SPI_DIRECTION_2LINES,
-        .DataSize = SPI_DATASIZE_16BIT,
-        .CLKPolarity = mode_ == MODE_SPI_ABS_AEAT ? SPI_POLARITY_HIGH : SPI_POLARITY_LOW,
-        .CLKPhase = SPI_PHASE_2EDGE,
-        .NSS = SPI_NSS_SOFT,
-        .BaudRatePrescaler = SPI_BAUDRATEPRESCALER_32,
-        .FirstBit = SPI_FIRSTBIT_MSB,
-        .TIMode = SPI_TIMODE_DISABLE,
-        .CRCCalculation = SPI_CRCCALCULATION_DISABLE,
-        .CRCPolynomial = 10,
-    };
-
-    if(mode_ & MODE_FLAG_ABS){
+    if(mode_ & MODE_FLAG_ABS) {
         abs_spi_cs_pin_init();
-
         if (axis_->controller_.config_.anticogging.pre_calibrated) {
             axis_->controller_.anticogging_valid_ = true;
         }
+    } else {
+        HAL_TIM_Encoder_Start(timer_, TIM_CHANNEL_ALL);
+        set_idx_subscribe();
+    }
+    
+    if(mode_ == MODE_SPI_ABS_TLE) {
+        spi_task_.config = {
+            .Mode = SPI_MODE_MASTER,
+            .Direction = SPI_DIRECTION_1LINE,
+            .DataSize = SPI_DATASIZE_16BIT,
+            .CLKPolarity = SPI_POLARITY_LOW,
+            .CLKPhase = SPI_PHASE_2EDGE,
+            .NSS = SPI_NSS_SOFT,
+            .BaudRatePrescaler = SPI_BAUDRATEPRESCALER_8,
```

Comment:
I'm slightly concerned about running at a higher clock rate due to how much trouble people have with the lower clock rate and other SPI encoders, but it works on my test bench and it seems to be beneficial for your use case so I'm just noting this down.

Reply:
Interesting... I didn't know people were having some trouble with the low clock rates on other SPI encoders. Do you know what sort of conditions they were running the encoders in? Using plain jumper cables right next to huge Motors and high power cables is probably not a good idea... twisted pairs with ground or shielding would help I bet.

Reply:
This is actually a big enough issue that we're in the process of releasing a differential SPI board to handle it. It seems to depend on the person and the encoder - some people have no problems with twisted pairs of some length, other people can't get reliable results even at short lengths.